crates/aptos/src/common/types.rs

```diff
@@ -1683,62 +1739,211 @@ pub struct ArgWithTypeVec {
-                .map_err(|err| CliError::UnableToParse("--type-args", err.to_string()))?;
-            parsed_type_args.push(type_tag)
+    /// Get instance as if all fields passed from command line, parsing JSON input file if needed.
+    fn check_input_style(self) -> CliTypedResult<EntryFunctionArguments> {
+        if self.json_file.is_none() {
+            if self.function_id.is_none() {
+                Err(CliError::CommandArgumentError(
+                    "Must provide either function ID or JSON input file".to_string(),
+                ))
+            } else {
+                Ok(self)
+            }
+        } else {
+            let json =
+                parse_json_file::<EntryFunctionArgumentsJSON>(self.json_file.as_ref().unwrap())?;
+            Ok(EntryFunctionArguments {
+                function_id: Some(MemberId::from_str(&json.function_id)?),
+                type_arg_vec: TypeArgVec::try_from(&json.type_args)?,
+                arg_vec: ArgWithTypeVec::try_from(&json.args)?,
+                json_file: None,
+            })
         }
+    }
+}
+
+impl TryInto<EntryFunction> for EntryFunctionArguments {
+    type Error = CliError;
 
+    fn try_into(self) -> Result<EntryFunction, Self::Error> {
+        let entry_function_args = self.check_input_style()?;
+        let function_id = entry_function_args.function_id.unwrap();
         Ok(EntryFunction::new(
-            self.function_id.module_id,
-            self.function_id.member_id,
-            parsed_type_args,
-            args,
+            function_id.module_id,
+            function_id.member_id,
+            entry_function_args.type_arg_vec.try_into()?,
+            entry_function_args.arg_vec.try_into()?,
         ))
     }
 }
 
-/// Common options for interactions with a multisig account.
-#[derive(Clone, Debug, Parser, Serialize)]
-pub struct MultisigAccount {
-    /// The address of the multisig account to interact with.
-    #[clap(long, parse(try_from_str=crate::common::types::load_account_arg))]
-    pub(crate) multisig_address: AccountAddress,
+impl TryInto<MultisigTransactionPayload> for EntryFunctionArguments {
+    type Error = CliError;
+
+    fn try_into(self) -> Result<MultisigTransactionPayload, Self::Error> {
+        Ok(MultisigTransactionPayload::EntryFunction(self.try_into()?))
+    }
+}
+
+impl TryInto<Option<MultisigTransactionPayload>> for EntryFunctionArguments {
+    type Error = CliError;
+
+    fn try_into(self) -> Result<Option<MultisigTransactionPayload>, Self::Error> {
+        if self.function_id.is_none() && self.json_file.is_none() {
+            Ok(None)
+        } else {
+            Ok(Some(self.try_into()?))
+        }
+    }
+}
+
+impl TryInto<ViewRequest> for EntryFunctionArguments {
+    type Error = CliError;
+
+    fn try_into(self) -> Result<ViewRequest, Self::Error> {
+        let entry_function_args = self.check_input_style()?;
+        let function_id = entry_function_args.function_id.unwrap();
```

Comment:
Could you leave a comment explaining why this unwrap is safe?

Reply:
https://github.com/aptos-labs/aptos-core/blob/6aa74479f6b3f2b44f06cc0aea7efbd9acff1a51/crates/aptos/src/common/types.rs#L1869-L1871